### DISCUSSION TITLE
What is the easiest way to open/switch-to WebAgentDriver app on iOS?

### DISCUSSION THREAD

**Discussion by bwomsm1** [appium/appium#18731] [Q&A] [answered]
What is the easiest way to open/switch-to WebAgentDriver app on iOS?

**Comment by jlipps** [comment]
Why do you want to do this?

**Comment by bwomsm1** [comment]
@jlipps hi, I am trying to face the issue of "set_clipboard_text", if I get it correctly from this issue response: https://github.com/appium/appium/issues/18730, I need to switch to WebAgentDriver and to perform the copy text there, 

am I correct? 

Bottom line, what I am trying to do is to copy text to the clipboard in iOS test.

**Reply by jlipps** [reply]
Got it. I would say use `mobile: activateApp` with WDA's bundle id just like launching any other app. Maybe the bundle id is [`com.facebook.wda.runner`](https://github.com/appium/WebDriverAgent/blob/master/WebDriverAgentRunner/Info.plist) (@mykola-mokhnach can confirm)

**Reply by KazuCocoa** [reply]
The default one would be `com.facebook.WebDriverAgentRunner.xctrunner` but the name depends on your provisioning profile

**Reply by bwomsm1** [reply, marked as answer]
Thanks @KazuCocoa, using `com.facebook.WebDriverAgentRunner.xctrunner` as following solved my issue:
```
# open WebDriverAgentRunner app
self.driver_.activate_app('com.facebook.WebDriverAgentRunner.xctrunner')

# wait 5 seconds for WebDriverAgentRunner to launch
self.wait(5)

# copy text to clipboard when WebDriverAgentRunner is opened
self.driver_.set_clipboard_text(text)

# launch back my test application
self.activate_app()
```

Thanks all.

**Reply by jlipps** [reply]
OK great. Clearly I got the wrong bundle ID. @KazuCocoa can you link where the correct default bundle ID is in the WDA source?

**Reply by KazuCocoa** [reply]
https://github.com/appium/WebDriverAgent/blob/244b8b85e671a20f21991fc13e21678848058fad/lib/constants.js#L4 might help.
`.xctrunner` part is automatically added by Xcode, so it is not a configurable part.

**Reply by bwomsm1** [reply]
@KazuCocoa do we have a global define in python for `com.facebook.WebDriverAgentRunner.xctrunner`, for case that the ID will change one day?

**Reply by KazuCocoa** [reply]
Maybe no as Appium (until we must change it by legal reasons etc), or until Apple changes the naming. `.xctrunner` part started since Xcode 11 or around. Thus newer Xcode can potentially change the naming. The config depends on your provisioning configuration as well. For example, when you create a provisioning profile for `com.example.*` for real devices, you may change it to `com.example.WebDriverAgentRunner` etc, then the naming should be `com.example.WebDriverAgentRunner.xctrunner`. Thus the naming depends on how users define the bundle id and what Xcode version they use.

**Reply by KazuCocoa** [reply]
Btw, https://appium.github.io/appium-xcuitest-driver/4.30/execute-methods/#mobile-setpasteboard also may help for simulator only

**Comment by bwomsm1** [comment]
@jlipps @mykola-mokhnach I tried to activate other app while my app was running in 2 ways, and none of them really switched to the Runner app:

First:
```
cap = {
            'platformVersion': '16.5',
            'deviceName': 'qa_iphone_02',
            'udid': '00__my_udid__E',
            'bundleId': 'com.facebook.wda.runner',
            'xcodeOrgId': 'B__my_org_id_7',
            'xcodeSigningId': 'iPhone Developer',
            'automationName': 'XCUITest',
            'app': '......path to /WebDriverAgentRunner-Runner.app',
            'platformName': 'iOS'
        }
        webdriver.Remote(self.__appium_server_ip, cap)
        self.wait(5)
```

Second:
```
self.driver_.activate_app('com.facebook.wda.runner')
```

In addition, my code was not continuing after both calls
